packages/shared/src/common/retry.ts

```diff
@@ -69,7 +69,7 @@ export type GetTimeToWaitParams = {
  * @see https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/
  */
 export async function executeWithRetries<T>(
-  fn: () => Promise<T>,
+  fn: (attempt: number) => Promise<T>,
```

Comment:
Just exposing the attempt number to functions using `executeWithRetries` - it's backwards compatible and safe.